cmake/alpakaCommon.cmake

```diff
@@ -313,257 +313,146 @@ endif()
+            endif()
 
-                # avoids warnings on host-device signatured, default constructors/destructors
-                alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS -Xcudafe=--diag_suppress=esa_on_defaulted_function_ignored)
-
-                # avoids warnings on host-device signature of 'std::__shared_count<>'
-                if(CUDA_VERSION EQUAL 10.0)
-                    alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS -Xcudafe=--diag_suppress=2905)
-                elseif(CUDA_VERSION EQUAL 10.1)
-                    alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS -Xcudafe=--diag_suppress=2912)
-                elseif(CUDA_VERSION EQUAL 10.2)
-                    alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS -Xcudafe=--diag_suppress=2976)
-                endif()
+            if(ALPAKA_CUDA_KEEP_FILES STREQUAL ON)
+                alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:--keep>)
+            endif()
 
-                if(ALPAKA_CUDA_KEEP_FILES STREQUAL ON)
-                    alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS --keep)
-                endif()
+            option(ALPAKA_CUDA_SHOW_CODELINES "Show kernel lines in cuda-gdb and cuda-memcheck. If ALPAKA_CUDA_KEEP_FILES is enabled source code will be inlined in ptx." OFF)
+            if(ALPAKA_CUDA_SHOW_CODELINES)
+                alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:--source-in-ptx -lineinfo>)
 
-                option(ALPAKA_CUDA_SHOW_CODELINES "Show kernel lines in cuda-gdb and cuda-memcheck. If ALPAKA_CUDA_KEEP_FILES is enabled source code will be inlined in ptx." OFF)
-                if(ALPAKA_CUDA_SHOW_CODELINES)
-                    alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS --source-in-ptx -lineinfo)
-                    if(NOT MSVC)
-                        alpaka_set_compiler_options(DEVICE list CUDA_NVCC_FLAGS -Xcompiler=-rdynamic)
-                    endif()
+                # This is shaky - We currently don't have a way of checking for the host compiler ID.
+                # See https://gitlab.kitware.com/cmake/cmake/-/issues/20901
+                if(NOT MSVC)
+                    alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=-rdynamic>)
                 endif()
+                set(ALPAKA_CUDA_KEEP_FILES ON CACHE BOOL "activate keep files" FORCE)
             endif()
-
-            if(OpenMP_CXX_FOUND)
-                # correctly propagate OpenMP flags
-                # This can be removed once we support CMake's first class CUDA support.
-                target_compile_options(alpaka INTERFACE ${OpenMP_CXX_FLAGS})
-            endif()
-
-            target_link_libraries(alpaka INTERFACE ${CUDA_CUDART_LIBRARY})
-            target_include_directories(alpaka INTERFACE ${CUDA_INCLUDE_DIRS})
         endif()
+
+        target_link_libraries(alpaka INTERFACE CUDA::cudart)
```

Comment:
Is this required when nvcc is the CUDA compiler? (Also the next line)

Reply:
I encountered this as well with LLAMA. If you just enable the CUDA language, you get a compiler able to compile CUDA. But you will not automatically get the proper include directories set for the target. So e.g. you IDE cannot find `<cuda_runtime.h>`. nvcc might still find the headers without this line though.